CSS modules: CSS table (#33236)
mdn/content: files/en-us/web/css/css_table/index.md

The **CSS table** module helps you define how to lay out table data.

> This CSS module defines styles applicable to the HTML {{htmlelement("table")}} element, used to render tabular data. By default, tables are rendered as a two-dimensional grid of table cells lined up in a series of consecutive rows and columns, generated from the table structure and sized according to their content. This module enables defining the position of the table's {{htmlelement("caption")}}, if present.

Reviewer: ```suggestion
This CSS module defines styles applicable to the HTML {{htmlelement("table")}} element, which is used to render tabular data. By default, tables are rendered as a two-dimensional grid with cells lined up in a series of consecutive rows and columns. This layout is generated from the table structure and sized according to the content of the cells. This module also enables defining the position of the table's {{htmlelement("caption")}}, if present.
```

- {{cssxref("caption-side")}}
- {{cssxref("empty-cells")}}
- {{cssxref("table-layout")}}

## Guides

> - [CSS table layout](/en-US/docs/Learn/CSS/CSS_layout/Introduction#table_layout)

Reviewer: Like the other two, should we prefix this and the next also with "Learn: "

```suggestion
- [Learn: CSS table layout](/en-US/docs/Learn/CSS/CSS_layout/Introduction#table_layout)
```

>   - : An overview of CSS layouts, including setting table-related `display` values to enable using CSS table properties on any element.

Reviewer: ```suggestion
  - : An overview of CSS layouts, including how to set table-related `display` values to enable using CSS table properties on any element.
```

> - [Styling tables](/en-US/docs/Learn/CSS/Building_blocks/Styling_tables)

Reviewer: ```suggestion
- [Learn: Styling tables](/en-US/docs/Learn/CSS/Building_blocks/Styling_tables)
```

>   - : A building block guide to improving the appearance of HTML tables, covering table styling techniques.

Reviewer: ```suggestion
  - : A guide to improving the appearance of HTML tables, covering table styling techniques.
```

- [Learn: HTML table basics](/en-US/docs/Learn/HTML/Tables/Basics)

>   - : Introduction to HTML tables, including the HTML for rows and cells, headings, making cells span multiple columns and rows, and grouping cells in a column for styling purposes.

Reviewer: What do you think about moving both the HTML guides to "See also"?
```suggestion
  - : An introduction to HTML tables, including the HTML for creating rows and cells, headings, making cells span multiple columns and rows, and grouping cells in a column for styling purposes.
```

Author: i want to leave them here because few developers know how to write tables correctly, so prefer to leave them under guides in the hopes that they are read.

- [Learn: HTML table advanced features and accessibility](/en-US/docs/Learn/HTML/Tables/Advanced)

>   - : Advanced HTML table features including captions and grouping table rows into table head, body and footer sections — as well as looking at the accessibility of tables for visually impaired users.

Reviewer: What do you think about moving the link to the HTML guide to "See also"?
```suggestion
  - : A look at advanced HTML table features, including captions and grouping table rows into table head, body and footer sections — as well as looking at the accessibility of tables for visually impaired users.
```

## Related concepts

- {{cssxref("display")}} property
- {{cssxref("vertical-align")}} property
- {{cssxref("text-align")}} property

> - HTML table-related elements

Reviewer: The HTML element list can be moved to after the CSS property list. WDYT?
```suggestion
- HTML table-related elements:
```

Author: I think the HTML elements are more relevant to anyone using this spec's properties, but won't argue if you insist.

  - {{htmlelement("table")}}
  - {{htmlelement("caption")}}
  - {{htmlelement("colgroup")}}
  - {{htmlelement("col")}}
  - {{htmlelement("thead")}}
  - {{htmlelement("tbody")}}
  - {{htmlelement("tfoot")}}
  - {{htmlelement("tr")}}
  - {{htmlelement("th")}}
  - {{htmlelement("td")}}

> - [CSS box sizing](/en-US/docs/Web/CSS/CSS_box_sizing) module

Reviewer: ```suggestion
- [CSS box model](/en-US/docs/Web/CSS/CSS_box_model) module
```

Author: The original was correct. See https://github.com/mdn/content/pull/33182

  - {{cssxref("box-sizing")}}
  - {{cssxref("height")}}
  - {{cssxref("max-width")}}
  - {{cssxref("min-height")}}
  - {{cssxref("min-width")}}
  - {{cssxref("width")}}
  - {{cssxref("min-content")}} keyword

- [CSS backgrounds and borders](/en-US/docs/Web/CSS/CSS_backgrounds_and_borders) module
  - {{cssxref("border")}} shorthand
  - {{cssxref("border-width")}}
  - {{cssxref("border-style")}}
  - {{cssxref("border-color")}}

## Specifications

> {{Specifications}}

Reviewer: I think we should replace the `spec-urls` front matter key value with "https://drafts.csswg.org/css-tables-3/", the same spec link that you're also using in the note after the Specifications table

Author: See the note below: the spec is not ready for implementation, and has a big orange warning when you visit the page, so opting not to update the spec-url.

> **Note:** The [CSS Table Module Level 3](https://drafts.csswg.org/css-tables-3/) specification is being developed, but is not yet ready for implementation.

## See also

- [CSS display](/en-US/docs/Web/CSS/CSS_display) module
- [CSS grid layout](/en-US/docs/Web/CSS/CSS_grid_layout) module

> - [CSS flex layout](/en-US/docs/Web/CSS/CSS_flexible_box_layout) module

Reviewer: ```suggestion
- [CSS flexible box layout](/en-US/docs/Web/CSS/CSS_flexible_box_layout) module
```
